docs/csharp/lambda-expressions.md

```diff
@@ -1,27 +1,160 @@
+
+## Statement lambdas ##
+
+A statement lambda resembles an expression lambda except that the statement(s) is enclosed in braces:
+
+```cs
+(input parameters) => { statement; }
+```
+
+The body of a statement lambda can consist of any number of statements; however, in practice there are typically no more than two or three.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/statement1.cs#1)]
+
+Statement lambdas, like anonymous methods, cannot be used to create expression trees.
+
+## Async lambdas ##
+
+You can easily create lambda expressions and statements that incorporate asynchronous processing by using the [async](language-reference/keywords/async.md) and [await](language-reference/keywords/await.md) keywords. For example, the example calls a `ShowSquares` method that executes asynchronously.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/async1.cs#1)]
+
+For more information about how to create and use async methods, see [Asynchronous programming with async and await](programming-guide/concepts/async/asynchronous-programming-with-async-and-await.md).
+
+## Lambda expressions and tuples ##
+
+Starting with C# 7.0, the C# language provides built-in support for tuples. You can provide a tuple as an argument to a lambda expression, and your lambda expression can also return a tuple. In some cases, the C# compiler uses type inference to determine the types of tuple components. 
+
+You define a tuple by enclosing a comma-delimited list of its components in parentheses. The following example uses tuple with 5 components to pass a sequence of numbers to a lambda expression, which doubles each value and returns a tuple with 5 components that contains the result of the multiplications.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/tuples1.cs#1)]
+
+Ordinarily, the fields of a tuple are named `Item1`, `Item2`, etc. You can, however, define a tuple with named components, as the following example does.
+
+[!CODE [csSnippets.Lambdas](../../samples/snippets/csharp/concepts/lambda-expressions/tuples2.cs#1)]
+
+For more information on support for tuples in C#, see [C# Tuple types](tuples.md).
+
+## Lambdas with the standard query operators ##
+
+Many standard query operators have an input parameter whose type is one of the @System.Func%601 family of generic delegates. These delegates use type parameters to define the number and type of input parameters, and the return type of the delegate. `Func` delegates are very useful for encapsulating user-defined expressions that are applied to each element in a set of source data. For example, consider the @System.Func%601 delegate, whose syntax is:
```

Comment:
I would start this by saying "LINQ to Objects, among other implementations" have an input parameter..

That way this is distinguished from the IQueryable discussion later.